sklearn/cluster/bicluster.py

```diff
@@ -35,10 +35,19 @@ def _scale_normalize(X):
 
     """
     X = make_nonnegative(X)
-    row_diag = np.asarray(1.0 / np.sqrt(X.sum(axis=1))).squeeze()
-    col_diag = np.asarray(1.0 / np.sqrt(X.sum(axis=0))).squeeze()
+    row_diag = np.asarray(1.0 / np.sqrt(X.sum(axis=1)))
+    if row_diag.shape[0] != 1:
```

Comment:
Sorry I know these changes reduce readability but basically the reason I added this condition is ``squeeze`` was converting it into a scalar when there is only 1 sample. Please let me know if there is a better way.